### PR TITLE
Wait for payment to complete before returning

### LINKED ISSUE #364: auto poll paying invoice

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #364

### CODE REVIEW

**Review by TonyGiorgio — COMMENTED** [review]
So pretty much, as long as it's within 30s, the client will get the correct invoice response, otherwise if it times out, it needs to get it from the list or get call?

**Review by TonyGiorgio — APPROVED** [review]

**Comment by benthecarman** [comment]
> So pretty much, as long as it's within 30s, the client will get the correct invoice response, otherwise if it times out, it needs to get it from the list or get call?

Yep!

File: mutiny-core/src/nodemanager.rs [hunk 1]
```diff
@@ -123,6 +127,59 @@ impl From<Invoice> for MutinyInvoice {
     }
 }
 
+impl MutinyInvoice {
```
[the remark this answers is not in the document]
Comment (benthecarman) [reply mutiny-core/src/nodemanager.rs:130]: Good call on the test, realized we weren't saving the pubkey we were key-sending to, fixed that and added tests for converting invoice and keysend payment infos.

**Comment by TonyGiorgio** [comment]
I had to run `rustup target add wasm32-unknown-unknown --toolchain nightly-2023-04-16`, maybe put that in the README? I put in some instructions for nightly yesterday so maybe around there.

**Comment by benthecarman** [comment]
@TonyGiorgio added

**Comment by TonyGiorgio** [comment]
cool LGTM!

**Pull request merged**